File: programmers/.dist/JS_language/추억점수.js
function solution(name, yearning, photo) {
  // 이름과 점수를 객체 배열로 묶기
  let nameyear = [];
  for (let i = 0; i < name.length; i++) {
    nameyear.push({ name: name[i], year: yearning[i] });
  }

  let result = [];

  for (let p of photo) {
    let sum = 0;
    for (let person of p) {
      // nameyear 배열에서 해당 이름 찾기
      let found = nameyear.find((ny) => ny.name === person);
      if (found) {
        sum += found.year;
      }
    }
    result.push(sum);
  }

  return result;
}

File: Java/5월/05_28/추억점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {

        Map<String,Integer>map=new HashMap<String, Integer>();

        for(int i=0; i<name.length; i++){
          map.put(name[i], yearning[i]);
        }
        
        int [] answer = new int[photo.length];

        for(int i=0; i<photo.length; i++){
          int count=0;

          for(String person : photo[i]){
            //map.containseKey는 포함되었는지 여부확인
            if(map.containsKey(person)){ 
              //포함되어있다면 값 증가
              count+=map.get(person);
            }
          }
          answer[i] = count;
        }

        return answer;
    }
}